src/Game/UI/uiFadeProgress.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <heap/seadDisposer.h>
+
+namespace uking::ui {
+
+class FadeProgress {
+    SEAD_SINGLETON_DISPOSER(FadeProgress)
+    FadeProgress() = default;
+    ~FadeProgress() = default;
+
+public:
+    void init();
+    void reset();
+    void updateFadeScreen();
+    void setProgress(float value);
+    float getGaugeValue();
```

Comment:
Sorry, missed this earlier, but could you make this a const member function?
```suggestion
    float getGaugeValue() const;
```